code/modules/ruins/lavalandruin_code/techcult.dm

```diff
@@ -0,0 +1,592 @@
+	icon_state = "lungs-c-u2"
+	safe_toxins_max = 50
+	safe_co2_max = 50
+	safe_oxygen_min = 6
+	maxHealth = 3 * STANDARD_ORGAN_THRESHOLD
+	emp_vulnerability = 5
+
+	var/dose_available = TRUE
+
+	cold_level_1_threshold = 140
+	cold_level_2_threshold = 80
+	cold_level_3_threshold = 30
+	heat_level_1_threshold = 800
+	heat_level_2_threshold = 1600
+	heat_level_3_threshold = 2400
+
+/obj/item/organ/lungs/cybernetic/tier4/on_life()
+	. = ..()
+	if(dose_available && owner.health <= owner.crit_threshold)
+		used_dose()
+
+/obj/item/organ/lungs/cybernetic/tier4/proc/used_dose()
+	owner.reagents.add_reagent(/datum/reagent/medicine/salbutamol, 5)
+	dose_available = FALSE
+	addtimer(VARSET_CALLBACK(src, dose_available, TRUE), 2 MINUTES)
+
+// Reviver Implant PLUS
+/obj/item/organ/cyberimp/chest/reviver/plus
+	name = "Reviver implant PLUS"
+	desc = "This implant will attempt to heal you REALLY FAST if you lose consciousness. For the true warriors!"
+	implant_color = "#CC0605" //Cooler red
+
+/obj/item/organ/cyberimp/chest/reviver/plus/on_life()
+	if(reviving)
+		if(owner.stat == UNCONSCIOUS)
+			addtimer(CALLBACK(src, .proc/heal), 10) //Fast
+		else
+			cooldown = revive_cost + world.time
+			reviving = FALSE
+			to_chat(owner, "<span class='notice'>Your reviver implant shuts down and starts recharging. It will be ready again in [DisplayTimeText(revive_cost)].</span>")
+		return
+
+	if(cooldown > world.time)
+		return
+	if(owner.stat != UNCONSCIOUS)
+		return
+	if(owner.suiciding)
+		return
+
+	revive_cost = 0
+	reviving = TRUE
+	to_chat(owner, "<span class='notice'>You feel a faint buzzing as your reviver implant starts patching your wounds...</span>")
+
+/obj/item/organ/cyberimp/chest/reviver/plus/heal()
+	if(owner.getOxyLoss())
+		owner.adjustOxyLoss(-10) //REAL fast
+		revive_cost += 5
+	if(owner.getBruteLoss())
+		owner.adjustBruteLoss(-5)
+		revive_cost += 40
+	if(owner.getFireLoss())
+		owner.adjustFireLoss(-5)
+		revive_cost += 40
+	if(owner.getToxLoss())
+		owner.adjustToxLoss(-3)
+		revive_cost += 40
+
+//Advanced Surgical Toolset Implant
+/obj/item/organ/cyberimp/arm/surgery/plus
+	name = "advanced surgical toolset implant"
+	desc = "A set of advanced surgical tools hidden behind a concealed panel on the user's arm."
+	contents = newlist(/obj/item/retractor/advanced/augment, /obj/item/surgicaldrill/advanced/augment, /obj/item/scalpel/advanced/augment, /obj/item/surgical_drapes)
+
+/obj/item/organ/cyberimp/arm/surgery/plus/emag_act(mob/user)
+	if(!(locate(/obj/item/reagent_containers/borghypo/hacked/augment) in items_list))
+		to_chat(user, "<span class='notice'>You unlock and hack integrated hypospray located in [src]!</span>") //Oh god oh fuck
+		items_list += new /obj/item/reagent_containers/borghypo/hacked/augment(src)
```

Comment:
jesus christ what the fuck why

I mean, I guess fulp usually doesn't allow murderboning, so it'll be hard to abuse this TOO badly, but still..

Reply:
It's quite balanced.
Keep in mind the SLOW regeneration rate and total volume, which is 10 for each toxin.